ods_ci/tasks/Resources/Provisioning/Hive/provision.robot

```diff
@@ -150,43 +150,26 @@ Create Floating IPs
-    WHILE   True    limit=${hive_timeout}    on_limit_message=Hive Install ${hive_timeout} Timeout Exceeded    # robotcode: ignore
-        ${old_log_data} = 	Get File 	${install_log_file}
-        ${last_line_index} =    Get Line Count    ${old_log_data}
-        IF    ${use_cluster_pool}
-            ${pod} =    Oc Get    kind=Pod    namespace=${namespace}
-        ELSE
-            ${pod} =    Oc Get    kind=Pod    namespace=${namespace}
-            ...    label_selector=hive.openshift.io/cluster-deployment-name=${cluster_name}
-        END
-        TRY
-            ${new_log_data} =    Oc Get Pod Logs    name=${pod[0]['metadata']['name']}    container=installer   namespace=${namespace}
-        EXCEPT
-            # Hive container (OCP installer log) is not ready yet
-            Log To Console    .    no_newline=true
-            Sleep   10s
-            CONTINUE
-        END
-        # Print the new lines that were added to the installer log
-        @{new_lines} =    Split To Lines    ${new_log_data}    ${last_line_index}
-        ${lines_count} =    Get length    ${new_lines}
-        IF  ${lines_count} > 0
-            Create File    ${install_log_file}    ${new_log_data}
-            FOR    ${line}    IN    @{new_lines}
-                Log To Console    ${line}
-            END
-        ELSE
-            ${hive_pods_status} =    Run And Return Rc    oc get pod -n ${namespace} --no-headers | awk '{print $3}' | grep -v 'Completed'
-            IF    ${hive_pods_status} != 0
-                Log    All Hive pods in ${namespace} have completed    console=True
-                BREAK
-            END
-            Log To Console    .    no_newline=true
-            Sleep   10s
-        END
+    [Arguments]    ${pool_name}    ${namespace}    ${hive_timeout}=70m
+    ${label_selector} =    Set Variable    hive.openshift.io/cluster-deployment-name=${cluster_name}
+    IF    ${use_cluster_pool}
+        ${label_selector} =    Set Variable    hive.openshift.io/clusterpool-name=${pool_name}
+    END
+    ${label_selector}=    Catenate    SEPARATOR=    ${label_selector}    ,hive.openshift.io/job-type=provision
+    ${logs_cmd} =     Set Variable    oc logs -f -l ${label_selector} -n ${namespace}
+    Wait For Pods To Be Ready    label_selector=${label_selector}    namespace=${namespace}    timeout=5m
+    TRY
+        ${return_code}    ${output}    ${err} =    Run And Watch Command    ${logs_cmd}    timeout=${hive_timeout}
+    EXCEPT
+        Log To Console    ERROR: Check Hive Logs if present or you may have hit timeout ${hive_timeout}.
+    END
+    Should Be Equal As Integers    ${return_code}    ${0}
+    ${hive_pods_status} =    Run And Return Rc
+    ...    oc get pod -n ${namespace} --no-headers | awk '{print $3}' | grep -v 'Completed'
+    IF    ${hive_pods_status} != 0
+        Log    All Hive pods in ${namespace} have completed    console=True
     END
-    Should Contain    ${new_log_data}    install completed successfully
```

Comment:
@bdattoma If you remove this check, it will not verify that the installation was completed successfully, and it may happen that the pod was completed but the installation failed.

Can you try to use `Run And Watch Command` with the argument `output_should_contain=install completed successfully`

Reply:
> pod was completed but the installation failed.


I think the pod fails in case of any install error
I can try adding that arg

Reply:
done